editor/import_dock.cpp

```diff
@@ -685,12 +689,25 @@ void ImportDock::_reimport() {
 		}
 
 		config->save(params->paths[i] + ".import");
+
+		//if the list of resources to import includes the currently-edited object in the inspector, save its path so we can load its reimported version later.
+		Ref<Resource> res = ResourceLoader::load(params->paths[i]);
+		if (*res == inspector_current_object) {
+			reload_inspector_resource_path = params->paths[i];
+		}
 	}
 
 	EditorFileSystem::get_singleton()->reimport_files(params->paths);
 	EditorFileSystem::get_singleton()->emit_signal(SNAME("filesystem_changed")); //it changed, so force emitting the signal
 
 	_set_dirty(false);
+
+	//load the reimported version of the edited resource and push it to EditorNode
+	//this needs to be deferred because the scene tree will also try to regain focus after reimport
```

Comment:
```suggestion
	// Load the reimported version of the edited resource and push it to EditorNode.
	// This needs to be deferred because the scene tree will also try to regain focus after reimport.
```